### PR TITLE
Add qt/Q Variable in Gen Histmaker, will not change anything else

### PR DESCRIPTION
**Pull request by tanmayvb**
This PR primarily adds the boson qt bY Q variable. An option is included to select the axis ptqVgen. Subsequently, if-else conditions are implemented to store kinematics and calculate Ai with respect to this variable. Otherwise, it will proceed with ptVgen as is. Additionally, the NNPDF30 branch name has been fixed. I trust this modification will not impact or interfere with any other aspects of the code.

### CODE REVIEW

**Comment by davidwalter2** [comment]
Can you say a bit more about the NNPDF branch? Did we use the wrong branch in the past? This would effect the pdf bias tests right?

**Comment by tanmayvb** [comment]
> Can you say a bit more about the NNPDF branch? Did we use the wrong branch in the past? This would effect the pdf bias tests right?
Yes, when I attempted to fetch all 100 variations for NNPDF30 with the AltSet-LHEPdfWeightAltSet13 branch, my local code broke after the first 50 variations. This is something we need to use for our theory-agnostic study as one of the alternate set, where we require all PDF variations. Upon closer inspection of the actual branch, I discovered that it was pointing to the wrong branch. I have attached a screenshot for your reference.

Branch-Name-PDF

File: scripts/histmakers/w_z_gen_dists.py [hunk 1]
```diff
@@ -119,13 +125,21 @@ def build_graph(df, dataset):
     df = theory_tools.define_theory_weights_and_corrs(df, dataset.name, corr_helpers, args)
 
     if isZ:
-        nominal_axes = [axis_massZgen, axis_rapidity, axis_ptVgen, axis_chargeZgen]
+        if(args.ptqVgen):
```
Comment (kdlong) [inline scripts/histmakers/w_z_gen_dists.py:128]: from a style perspective ```if args.ptVgen:``` is preferable

File: scripts/histmakers/w_z_gen_dists.py [hunk 2]
```diff
@@ -237,7 +251,10 @@ def build_graph(df, dataset):
     return results, weightsum
 
 resultdict = narf.build_and_run(datasets, build_graph)
-output_tools.write_analysis_output(resultdict, f"{os.path.basename(__file__).replace('py', 'hdf5')}", args, update_name=not args.forceDefaultName)
+if(args.ptqVgen):
+    output_tools.write_analysis_output(resultdict, f"{os.path.basename(__file__).replace('.py', f'_pdf_{args.pdfs[0]}_vars_qtbyQ.hdf5')}", args, update_name=not args.forceDefaultName)
```
[the remark this answers is not in the document]
Comment (tanmayvb) [reply scripts/histmakers/w_z_gen_dists.py]: Yes, moved within write_analysis_output

**Comment by kdlong** [comment]
If this is only relevant for the w_z_gen_dists.py script it should not go in common. Otherwise it looks good to me

**Comment by tanmayvb** [comment]
> If this is only relevant for the w_z_gen_dists.py script it should not go in common. Otherwise it looks good to me

Hi Kenneth, then can we do something like that in write_analysis_output function **if w_z_gen_dists in outfile and  if args.ptqVgen to_append.append("vars_qtbyQ")**

File: utilities/io_tools/output_tools.py [hunk 3]
```diff
@@ -58,6 +58,10 @@ def write_analysis_output(results, outfile, args, update_name=True):
         to_append.append(args.theoryCorr[0]+"Corr")
     if args.maxFiles is not None:
         to_append.append(f"maxFiles_{args.maxFiles}".replace("-","m"))
+    if args.pdfs and args.pdfs != ["msht20"]:
+        to_append.append(args.pdfs[0])
+    if args.ptqVgen:
```
Comment (kdlong) [inline utilities/io_tools/output_tools.py:63]: if hasattr(args, "ptqVgen") and args.ptqVgen:

**Comment by kdlong** [comment]
> > If this is only relevant for the w_z_gen_dists.py script it should not go in common. Otherwise it looks good to me
> 
> Hi Kenneth, then can we do something like that in write_analysis_output function **if w_z_gen_dists in outfile and if args.ptqVgen to_append.append("vars_qtbyQ")**

Ok I see why you did this then. You can check that the variable has been defined with hasattr. I put a comment in the code.

File: utilities/io_tools/output_tools.py [hunk 4]
```diff
@@ -58,6 +58,10 @@ def write_analysis_output(results, outfile, args, update_name=True):
         to_append.append(args.theoryCorr[0]+"Corr")
     if args.maxFiles is not None:
         to_append.append(f"maxFiles_{args.maxFiles}".replace("-","m"))
+    if args.pdfs and args.pdfs != ["msht20"]:
```
Comment (kdlong) [inline utilities/io_tools/output_tools.py:61]: You might want the check to be args.pdfs[0] != "msht20" instead

Comment (tanmayvb) [reply utilities/io_tools/output_tools.py]: Thank you Kenneth and done now

File: utilities/io_tools/output_tools.py (hunk 3 above)
Comment (tanmayvb) [reply utilities/io_tools/output_tools.py]: Done

**Pull request merged**